docs/topics/wait_for.rst

```diff
@@ -0,0 +1,80 @@
+:orphan:
+.. currentmodule:: discord
+
+.. _guide_wait_for:
+
+Waiting For Events
+=============================
+When creating commands or handling events, you often find yourself wanting to wait for an action or user response. In discord.py, we use the :meth:`Client.wait_for` method. 
+
+What is ``wait_for``?
+~~~~~~~~~~~~~~~~~~~~~
+:meth:`Client.wait_for` is similar to client.event/listen, however instead of calling a function, it instead holds execution of your code until the event has been dispatched, and returns the data that was in the event. 
+
+
+Here is a quick example
+
+.. code-block:: python3
+   :emphasize-lines: 2
+
+    await channel.send('say hi!')
+    message = await client.wait_for('message') # wait for a message, the wait_for will return the message it finds
+    await message.reply('hello!')
+
+The key line here is line 2. We are waiting for an event to happen, in this case, a message event.
+
+Wait_for is commonly used to control flow in a code, waiting for message replies or reactions to be added. You can wait for any event you want.
+
+Checks and Timeouts
+~~~~~~~~~~~~~~~~~~~~
+While the example above allows us to wait for an event, this will pass as soon as *any* event of that type is dispatched. Sometimes, we want to filter it. To do this, we use the check keyword argument.
```

Comment:
```suggestion
While the example above allows us to wait for an event, this will pass as soon as *any* event of that type is dispatched. Sometimes, we want to filter it. To do this, we use the ``check`` keyword argument.
```